gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -3,38 +3,52 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.gcloud.storage.contrib.nio.CloudStorageFileSystem.FILE_TIME_UNKNOWN;
 
-import com.google.appengine.tools.cloudstorage.GcsFileMetadata;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Optional;
 import com.google.common.collect.ImmutableMap;
+import com.google.gcloud.storage.Acl;
+import com.google.gcloud.storage.BlobInfo;
 
 import java.nio.file.attribute.FileTime;
+import java.util.List;
 import java.util.Objects;
 
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
-/** Metadata for a Google Cloud Storage object. */
+/**
+ * Metadata for a Google Cloud Storage object.
+ */
 @Immutable
 final class CloudStorageObjectAttributes implements CloudStorageFileAttributes {
 
-  private final GcsFileMetadata metadata;
+  private final BlobInfo info;
+  private final String name;
 
-  CloudStorageObjectAttributes(GcsFileMetadata metadata) {
-    this.metadata = checkNotNull(metadata);
+  CloudStorageObjectAttributes(BlobInfo info) {
+    this.info = checkNotNull(info);
+    this.name = info.name();
+  }
+
+  CloudStorageObjectAttributes(String directoryName) {
+    this.info = null;
+    this.name = checkNotNull(directoryName);
   }
 
   @Override
   public long size() {
-    return metadata.getLength();
+    if (null == info) {
+      return 0;
+    }
+    return info.size();
   }
 
   @Override
   public FileTime creationTime() {
-    if (metadata.getLastModified() == null) {
+    if (null == info || info.updateTime() == null) {
```

Comment:
Cloud storage doesn't have folders. Are you talking about pseudo directories? Because the behavior for performing IO on pseudo directories is to throw [CloudStoragePseudoDirectoryException](https://github.com/jean-philippe-martin/gcloud-java/blob/gcs-nio/gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStoragePseudoDirectoryException.java)